--- /src/main/java/com/fasterxml/jackson/databind/deser/BuilderBasedDeserializer.java	2022-05-05 00:27:45.103675427 -0400
+++ /src/main/java/com/fasterxml/jackson/databind/deser/BuilderBasedDeserializer.java	2022-05-05 00:30:06.041927408 -0400
@@ -580,6 +580,8 @@
             // creator property?
             SettableBeanProperty creatorProp = creator.findCreatorProperty(propName);
             if (creatorProp != null) {
+if (buffer.assignParameter(creatorProp,creatorProp.deserialize(p,ctxt)))  
+
                 if (buffer.assignParameter(creatorProp, creatorProp.deserialize(p, ctxt))) {
                     t = p.nextToken();
                     Object bean;

