--- /src/main/java/com/fasterxml/jackson/databind/ser/PropertyBuilder.java	2022-05-03 16:28:43.258371402 -0400
+++ /src/main/java/com/fasterxml/jackson/databind/ser/PropertyBuilder.java	2022-05-03 16:40:29.541634307 -0400
@@ -135,7 +135,8 @@
 
         // property annotation override
         
-        inclV = inclV.withOverrides(propDef.findInclusion());
+        suppressNulls = _defaultInclusion.getValueInclusion() == JsonInclude.Include.NON_DEFAULT;
+inclV = inclV.withOverrides(propDef.findInclusion());
         JsonInclude.Include inclusion = inclV.getValueInclusion();
 
         if (inclusion == JsonInclude.Include.USE_DEFAULTS) { // should not occur but...

