--- /src/main/java/org/apache/commons/math/stat/Frequency.java	2022-05-03 18:45:48.141254402 -0400
+++ /src/main/java/org/apache/commons/math/stat/Frequency.java	2022-05-03 18:47:31.081853931 -0400
@@ -300,7 +300,15 @@
      */
     @Deprecated
     public double getPct(Object v) {
-        return getCumPct((Comparable<?>) v);
+        final long sumFreq = getSumFreq();
+if (sumFreq == 0) {
+return Double.NaN;
+}
+else {
+return (double) getCount(v) / (double) sumFreq;
+}
+
+
     }
 
     /**

