--- /src/java/org/apache/commons/jxpath/ri/model/dom/DOMNodePointer.java	2022-05-04 08:02:51.757302598 -0400
+++ /src/java/org/apache/commons/jxpath/ri/model/dom/DOMNodePointer.java	2022-05-04 08:10:54.203376058 -0400
@@ -412,7 +412,8 @@
         Element element = (Element) node;
         String prefix = name.getPrefix();
         if (prefix != null) {
-            String ns = getNamespaceURI(prefix);
+            String ns = getNamespaceResolver().getNamespaceURI(prefix);
+
             if (ns == null) {
                 throw new JXPathException(
                     "Unknown namespace prefix: " + prefix);

