--- /src/main/java/org/apache/commons/math/optimization/direct/MultiDirectional.java	2022-05-02 17:32:45.738495941 -0400
+++ /src/main/java/org/apache/commons/math/optimization/direct/MultiDirectional.java	2022-05-02 17:36:47.301250940 -0400
@@ -87,7 +87,7 @@
 
             // compute the contracted simplex
             final RealPointValuePair contracted = evaluateNewSimplex(original, gamma, comparator);
-            if (comparator.compare(contracted, best) < 0) {
+            if (comparator.compare(contracted, reflected) <= 0) {
                 // accept the contracted simplex
 
             // check convergence

