--- /src/java/org/apache/commons/cli/PosixParser.java	2022-05-03 02:17:11.914860804 -0400
+++ /src/java/org/apache/commons/cli/PosixParser.java	2022-05-03 02:22:33.910466741 -0400
@@ -299,7 +299,8 @@
             }
             else if (stopAtNonOption)
             {
-                process(token.substring(i));
+                process(token.substring(i));return ;
+
             }
             else
             {

