--- /src/java/org/apache/commons/jxpath/ri/compiler/CoreOperationRelationalExpression.java	2022-05-03 22:48:15.564406881 -0400
+++ /src/java/org/apache/commons/jxpath/ri/compiler/CoreOperationRelationalExpression.java	2022-05-04 00:09:28.643301977 -0400
@@ -39,7 +39,11 @@
     }
 
     public final Object computeValue(EvalContext context) {
-        return compute(args[0].computeValue(context), args[1]
+        if (!InfoSetUtil.booleanValue(args[0].computeValue(context))) {
+return Boolean.FALSE;
+}
+
+return compute(args[0].computeValue(context), args[1]
                 .computeValue(context)) ? Boolean.TRUE : Boolean.FALSE;
     }
 

