--- /src/main/java/org/joda/time/MonthDay.java	2022-05-03 18:27:53.787079573 -0400
+++ /src/main/java/org/joda/time/MonthDay.java	2022-05-03 18:42:11.522459261 -0400
@@ -516,7 +516,8 @@
             return this;
         }
         int[] newValues = getValues();
-        newValues = getField(index).add(this, index, newValues, amount);
+        newValues = getField(index).addWrapPartial(this, index, newValues, amount);
+
         return new MonthDay(this, newValues);
     }
 

