--- /src/main/java/org/jsoup/parser/HtmlTreeBuilder.java	2022-05-02 20:23:11.057628590 -0400
+++ /src/main/java/org/jsoup/parser/HtmlTreeBuilder.java	2022-05-02 20:27:31.877152580 -0400
@@ -391,7 +391,7 @@
             if ("select".equals(name)) {
                 transition(HtmlTreeBuilderState.InSelect);
                 break; // frag
-            } else if (("td".equals(name) || "td".equals(name) && !last)) {
+            } else if (!last) {
                 transition(HtmlTreeBuilderState.InCell);
                 break;
             } else if ("tr".equals(name)) {

