--- /src/main/java/com/fasterxml/jackson/core/filter/FilteringParserDelegate.java	2022-05-03 07:56:17.806901181 -0400
+++ /src/main/java/com/fasterxml/jackson/core/filter/FilteringParserDelegate.java	2022-05-03 08:01:30.580652541 -0400
@@ -237,7 +237,7 @@
             //if not scalar and ended successfully, and !includePath, then return null
                 if (_currToken.isStructEnd()) {
                     if (_headContext.isStartHandled()) {
-                        return (_currToken = null);
+                        
                     }
                 } else if (_currToken.isScalarValue()) {
                     //else if scalar, and scalar not present in obj/array and !includePath and INCLUDE_ALL matched once

