--- /src/main/java/org/jsoup/parser/HtmlTreeBuilder.java	2022-05-02 21:26:28.951672388 -0400
+++ /src/main/java/org/jsoup/parser/HtmlTreeBuilder.java	2022-05-02 21:27:17.337079384 -0400
@@ -467,7 +467,7 @@
         // https://html.spec.whatwg.org/multipage/parsing.html#has-an-element-in-the-specific-scope
         int bottom = stack.size() -1;
         if (bottom > MaxScopeSearchDepth) {
-            bottom = MaxScopeSearchDepth;
+            
         }
         final int top = bottom > MaxScopeSearchDepth ? bottom - MaxScopeSearchDepth : 0;
         // don't walk too far up the tree

