--- /src/main/java/org/apache/commons/compress/archivers/tar/TarUtils.java	2022-05-04 00:21:48.074998589 -0400
+++ /src/main/java/org/apache/commons/compress/archivers/tar/TarUtils.java	2022-05-04 00:29:07.125063019 -0400
@@ -132,9 +132,7 @@
 
         for ( ;start < end; start++) {
             final byte currentByte = buffer[start];
-            if (currentByte == 0) {
-                break;
-            }
+            
             // CheckStyle:MagicNumber OFF
             if (currentByte < '0' || currentByte > '7'){
                 throw new IllegalArgumentException(

