--- /src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveOutputStream.java	2022-05-02 16:41:41.301389881 -0400
+++ /src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveOutputStream.java	2022-05-02 16:42:52.532306641 -0400
@@ -384,10 +384,7 @@
     @Override
     public void write(byte[] wBuf, int wOffset, int numToWrite) throws IOException {
         if ((currBytes + numToWrite) > currSize) {
-            throw new IOException("request to write '" + numToWrite
-                                  + "' bytes exceeds size in header of '"
-                                  + currSize + "' bytes for entry '"
-                                  + currName + "'");
+            
 
             //
             // We have to deal with assembly!!!

