diff --git a/src/main/java/org/apache/commons/compress/archivers/tar/TarUtils.java b/src/main/java/org/apache/commons/compress/archivers/tar/TarUtils.java
index 4940d1d5..d3ca809a 100644
--- a/src/main/java/org/apache/commons/compress/archivers/tar/TarUtils.java
+++ b/src/main/java/org/apache/commons/compress/archivers/tar/TarUtils.java
@@ -183,12 +183,19 @@ public class TarUtils {
     private static long parseBinaryLong(final byte[] buffer, final int offset,
                                         final int length,
                                         final boolean negative) {
+// start of generated patch
+if((offset+length)>=9){
+throw new IllegalArgumentException("At offset "+offset);
+}
+// end of generated patch
+/* start of original code
         if (length >= 9) {
             throw new IllegalArgumentException("At offset " + offset + ", "
                                                + length + " byte binary number"
                                                + " exceeds maximum signed long"
                                                + " value");
         }
+ end of original code*/
         long val = 0;
         for (int i = 1; i < length; i++) {
             val = (val << 8) + (buffer[offset + i] & 0xff);
