--- /src/main/java/org/apache/commons/math/analysis/solvers/BisectionSolver.java	2022-05-03 06:59:09.323441812 -0400
+++ /src/main/java/org/apache/commons/math/analysis/solvers/BisectionSolver.java	2022-05-03 07:56:17.576523032 -0400
@@ -69,7 +69,8 @@
     /** {@inheritDoc} */
     public double solve(final UnivariateRealFunction f, double min, double max, double initial)
         throws MaxIterationsExceededException, FunctionEvaluationException {
-        return solve(min, max);
+        this.f = f;
+return solve(min, max);
     }
 
     /** {@inheritDoc} */
